### ISSUE TITLE
[Engine] Should be able to scale global timestep

### ISSUE THREAD

**Issue by mattbenic** [AtomicGameEngine/AtomicGameEngine#1413]
There are scenarios in which the global timestep would need to be scaled, as an example when all animations should be played in slow motion. I'd suggest modifying the update message to include the actual timestep, scaled timestep, user specified scale and any other useful timing information. 

Current subscribers to this message would need to be updated to use the scaled value.

**Comment by JoshEngebretson** [comment]
I think the event data having the time scaling info is good, avoids getting the Time subsystem to read it.  

One note, In C# components, there is an optimization that passes the current timeStep as a float into Update instead of fiddling with the variant map event data.  In C#/JS I think we will want the passed in float timeStep to be scaled, with the Time subsystem in C#/JS holding unscaled value.  This is for script logic and will cover most (99%?) of the cases

**Comment by Alan-FGR** [comment]
Yeah, I agree it makes sense to pass the scaled delta of time in Update() and give access to a the real change in time directly in the Time subsystem. What matters for the game logic is the actual time changes in the game itself.
